api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/MedicationQuantityCodingTranslatorImpl.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api.translators.impl;
+
+import javax.annotation.Nonnull;
+
+import lombok.AccessLevel;
+import lombok.Setter;
+import org.hl7.fhir.r4.model.CodeableConcept;
+import org.hl7.fhir.r4.model.Coding;
+import org.hl7.fhir.r4.model.ICoding;
+import org.openmrs.Concept;
+import org.openmrs.module.fhir2.FhirConstants;
+import org.openmrs.module.fhir2.api.translators.CodingTranslator;
+import org.openmrs.module.fhir2.api.translators.ConceptTranslator;
+import org.openmrs.util.OpenmrsUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+@Setter(AccessLevel.PACKAGE)
+/**
+ * This is an implementation of Coding Translator that maps a Medication Quantity Concept to/from a
+ * Coding in FHIR with a preferred set of systems and codes to prioritize. This will first favor
+ * using RxNorm as the coding system. If this is not present on the OpenMRS concept, it will next
+ * favor SNOMED-CT. Finally, if neither are present, it will favor the Concept UUID with a null
+ * system.
+ */
+public class MedicationQuantityCodingTranslatorImpl implements CodingTranslator {
```

Comment:
I'm wondering if we want to call this "SimpleQuantityCodingTranslator" or even just "BaesCodingTranslator" or "CodingTranslator" until we determine that what we want do here is specific to medication and need to do something different for the coding of another simple quantity?

Or are there already cases/examples in the code where we already do something differnet?

In particular, it was a little confusing to me because my assumption based on the name would be that this translator was only used for translating the quantity of medication ordered/dispensed, and not also the dose (whicih, of course is "a quantity associated with a medication", but wasn't my first reading of the meaning of the class name).

Reply:
@mogoodrich I added an alternative idea on the ticket, but on reflection I think I'm in favor of keeping this as is, pending @ibacher 's thoughts.

Reply:
Makes sense, I adding my thoughts on the ticket.